geopandas/tests/test_sindex.py

```diff
@@ -86,9 +82,21 @@ def test_polygons_append(self):
 
     def test_lazy_build(self):
         s = GeoSeries([Point(0, 0)])
-        assert s._sindex is None
+        assert s.values._sindex is None
         assert s.sindex.size == 1
-        assert s._sindex is not None
+        assert s.values._sindex is not None
+
+    def test_rebuild_on_item_change(self):
+        s = GeoSeries([Point(0, 0)])
+        original_index = s.sindex
+        s.iloc[0] = Point(0, 0)
+        assert s.sindex is not original_index
+
+    def test_rebuild_on_slice(self):
+        s = GeoSeries([Point(0, 0), Point(0, 0)])
+        original_index = s.sindex
+        sliced = s.iloc[:1]
+        assert sliced.sindex is not original_index
```

Comment:
Can you add something similar for GeoDataFrame? 

And maybe also a test (if we don't have that yet) that actually checks that for certain operations, we *keep* the same sindex (eg now it is moved to the array, selecting the geometry column of a GeoDataFrame can preserve the sindex I think?)

Reply:
Added both!